app/controllers/posts_controller.rb

```diff
@@ -4,7 +4,34 @@ def index
     @user_posts = @user.posts
   end
 
+  def new
+    @id = current_user.id
+    post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: } }
+    end
+  end
+
+  def create
+    @id = current_user.id
+    post = Post.new(title: params[:title], text: params[:text], user: current_user, comments_counter: 0,
+                    likes_counter: 0)
+    respond_to do |format|
+      format.html do
+        if post.save
+          post.counter_updater
+          flash[:success] = 'Post saved successfully'
+          redirect_to user_posts_path
+        else
+          flash.now[:error] = 'Error: Post could not be saved'
+          render :new, locals: { post: }
+        end
+      end
+    end
+  end
+
   def show
-    @user_post = Post.find_by(user_id: params[:user_id], id: params[:id])
+    @user_post = Post.find(params[:id])
+    @user_id = current_user.id
   end
```

Comment:
- [ ] [Optional] Please use strong parameters in this [lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.) and that's to prevent any unwanted parameters added when creating a new post.